src/app/shared/navigation.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { Location } from '@angular/common';
+import { NavigationEnd, Router } from '@angular/router';
+
+/**
+ * See https://nils-mehlhorn.de/posts/angular-navigate-back-previous-page/
+ */
+@Injectable({ providedIn: 'root' })
+export class NavigationService {
+  private history: string[] = [];
+
+  constructor(private router: Router, private location: Location) {
+    this.router.events.subscribe((event) => {
+      if (event instanceof NavigationEnd) {
+        this.history.push(event.urlAfterRedirects);
+        this.history.splice(0, this.history.length - 5); // only keep the latest few, no need for more.
+      }
+    });
+  }
+
+  back(): void {
+    const record = this.history.pop();
+    if (this.history.length > 0) {
+      this.location.back();
```

Comment:
It would be nice to have a `back` configured. So you could use something like `router.navigate(this.back)`.